src/jbuild.ml

```diff
@@ -762,19 +772,51 @@ module Rule = struct
          field "deps"    (list Dep_conf.t) ~default:[] >>= fun deps ->
          field "action"  Action.Unexpanded.t           >>= fun action ->
          field "locks"   (list String_with_vars.t) ~default:[] >>= fun locks ->
-         field_b "fallback" >>= fun fallback ->
+         map_validate
+           (field_b "fallback" >>= fun fallback ->
+            field_o "mode" Mode.t >>= fun mode ->
+            return (fallback, mode))
+           ~f:(function
+             | true, Some _ ->
+               Error "Cannot use both (fallback) and (mode ...) at the same time.\n\
+                      (fallback) is the same as (mode fallback), \
+                      please use the latter in new code."
```

Comment:
Will `(fallback)` be removed in 1.0?

Reply:
yes, we'll get the upgrade tool to translate it to `(mode fallback)`